src/deposit.cairo

```diff
@@ -21,28 +24,43 @@ mod Deposit {
     };
 
     use starknet::{
-        ContractAddress, get_contract_address, get_tx_info, event::EventEmitter,
-        storage::{StoragePointerWriteAccess, StoragePointerReadAccess}
+        ContractAddress, ClassHash, get_contract_address, get_caller_address, get_tx_info,
+        event::EventEmitter, storage::{StoragePointerWriteAccess, StoragePointerReadAccess}
     };
 
+    component!(
+        path: ReentrancyGuardComponent, storage: reentrancy_guard, event: ReentrancyGuardEvent
+    );
+    component!(path: UpgradeableComponent, storage: upgradeable, event: UpgradeableEvent);
+
+    impl ReentrancyInternalImpl = ReentrancyGuardComponent::InternalImpl<ContractState>;
+    impl UpgradeableInternalImpl = UpgradeableComponent::InternalImpl<ContractState>;
+
     #[storage]
     struct Storage {
         owner: ContractAddress,
         ekubo_core: ICoreDispatcher,
         zk_market: IMarketDispatcher,
-        is_position_open: bool
+        treasury: ContractAddress,
+        is_position_open: bool,
+        #[substorage(v0)]
+        reentrancy_guard: ReentrancyGuardComponent::Storage,
+        #[substorage(v0)]
+        upgradeable: UpgradeableComponent::Storage
     }
 
     #[constructor]
     fn constructor(
         ref self: ContractState,
         owner: ContractAddress,
         ekubo_core: ICoreDispatcher,
-        zk_market: IMarketDispatcher
+        zk_market: IMarketDispatcher,
+        treasury: ContractAddress
     ) {
         self.owner.write(owner);
```

Comment:
Can we assert that the owner is nonzero here? Because otherwise anyone can upgrade the contract (easy to spoof zero tx sender address, such as when pushing from L1)